ui/app/src/model/project-client.ts

```diff
@@ -98,38 +158,45 @@ export function useAddProjectMutation() {
  */
 export function useDeleteProjectMutation() {
   const queryClient = useQueryClient();
-  const key = buildQueryKey({ resource });
-
-  return useMutation<string, Error, string>({
-    mutationKey: key,
-    mutationFn: (name: string) => {
-      const url = buildURL({ resource, name });
-      return fetch(url, {
-        method: HTTPMethodDELETE,
-        headers: HTTPHeader,
-      }).then(() => {
-        return name;
-      });
+  const queryKey = buildQueryKey({ resource });
+
+  return useMutation<ProjectResource, Error, ProjectResource>({
+    mutationKey: queryKey,
+    mutationFn: async (entity: ProjectResource) => {
+      await deleteProject(entity);
+      return entity;
     },
-    onSuccess: (name) => {
-      queryClient.removeQueries([...key, name]);
+    onSuccess: (entity: ProjectResource) => {
+      queryClient.removeQueries([...queryKey, entity.metadata.name]);
 
       const dependingKeys = dependingResources.map((resource) => buildQueryKey({ resource }));
-      dependingKeys.forEach((k) => queryClient.removeQueries([...k, name]));
+      dependingKeys.forEach((k) => queryClient.removeQueries([...k, entity.metadata.name]));
 
       return Promise.all([
         ...dependingKeys.map((k) => queryClient.invalidateQueries(k)),
         queryClient.invalidateQueries([userKey]),
-        queryClient.invalidateQueries(key),
+        queryClient.invalidateQueries(queryKey),
       ]);
     },
   });
 }
 
-export function getProject(name: string) {
-  const url = buildURL({ resource: resource, name: name });
-  return fetchJson<ProjectResource>(url, {
-    method: HTTPMethodGET,
-    headers: HTTPHeader,
+async function dashboardsPerProject(): Promise<ProjectDashboardsGroup[]> {
```

Comment:
I would have put it in dashboard personally

Reply:
I did that first, but circular deps 😄

Reply:
Arh I'm even the one that introduced that dep :D I remember now, nevermind